src/Serialization/upgrades/main.jl

```diff
@@ -0,0 +1,21 @@
+struct UpgradeScript
```

Comment:
I think some comments in this file would be really important. Of course if this is meant to be a draft for discussion, then not, but for the final version, some explanation as to what's going on here is essential. (Even for discussion, *some* comments at least would help)